src/main/java/org/sonarlint/intellij/actions/MarkAsResolvedAction.kt

```diff
@@ -0,0 +1,263 @@
+                displayErrorNotification(project, errorTitle, "Could not check status change permission", GROUP)
+                null
+            }
+        }
+
+        private fun confirm(project: Project, productName: String, issueStatus: IssueStatus): Boolean {
+            if (PropertiesComponent.getInstance().getBoolean(SKIP_CONFIRM_DIALOG_PROPERTY, false)) {
+                return true
+            }
+            return MessageDialogBuilder.okCancel(
+                "Confirm marking issue as resolved",
+                "Are you sure you want to mark this issue as \"${issueStatus.title}\"? The status will be modified on $productName"
+            )
+                .yesText("Confirm")
+                .noText("Cancel")
+                .doNotAsk(DoNotShowAgain())
+                .ask(project)
+        }
+
+        private fun serverConnection(project: Project): ServerConnection? = getService(
+            project,
+            ProjectBindingManager::class.java
+        ).tryGetServerConnection().orElse(null)
+    }
+
+    override fun isEnabled(e: AnActionEvent, project: Project, status: AnalysisStatus): Boolean {
+        return ((e.getData(DisableRuleAction.ISSUE_DATA_KEY) != null && e.getData(DisableRuleAction.ISSUE_DATA_KEY)?.serverFindingKey != null
+            && e.getData(DisableRuleAction.ISSUE_DATA_KEY)?.isValid == true) || (e.getData(TAINT_VULNERABILITY_DATA_KEY) != null))
+    }
+
+    override fun updatePresentation(e: AnActionEvent, project: Project) {
+        val serverConnection = serverConnection(project) ?: return
+        e.presentation.description = "Mark as Resolved on ${serverConnection.productName}"
+    }
+
+
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.project ?: return
+        val issue = e.getData(DisableRuleAction.ISSUE_DATA_KEY)
+        val vulnerability = e.getData(TAINT_VULNERABILITY_DATA_KEY)
+        val file: VirtualFile
+        var isTaintVulnerability: Boolean = false;
+        var issueKey: String
+
+        if (issue == null && vulnerability == null) {
+            return displayErrorNotification(project, errorTitle, "The issue could not be found.", GROUP)
+        }
+
+        if (issue != null) {
+            file = issue.file
+            issueKey =
+                issue.serverFindingKey ?: return displayErrorNotification(project, errorTitle, "The issue key could not be found.", GROUP)
+        } else {
+            file = vulnerability?.file() ?: return displayErrorNotification(project, errorTitle, "The file could not be found.", GROUP)
+            isTaintVulnerability = true
+            issueKey = vulnerability.key()
+        }
+
+        openMarkAsResolvedDialog(project, file, issueKey, isTaintVulnerability, issue, vulnerability)
+    }
+
+
+    private class DoNotShowAgain : DoNotAskOption {
+        override fun isToBeShown(): Boolean {
+            return true
+        }
```

Comment:
For all the following this syntax can be used

```suggestion
        override fun isToBeShown() = true
```